music21/stream/base.py

```diff
@@ -3709,7 +3709,71 @@ def getElementsByOffset(
             sIterator = sIterator.getElementsByClass(classList)
         return sIterator
 
-    def getElementAtOrBefore(self, offset, classList=None):
+    def getContextByClass(
+        self,
+        className,
+        *,
+        getElementMethod='getElementAtOrBefore',
+        sortByCreationTime=False,
+        followDerivation=True,
+        priorityTargetOnly=False,
+    ) -> Optional[base.Music21Object]:
+        '''
+        Overrides :meth:`~music21.base.Music21Object.getContextByClass` in order
+        to look first in this stream's elements at offset 0.0 before executing the
+        rest of the context search, so long as `getElementMethod` is the default
+        `getElementAtOrBefore`.
+
+        These `getElementMethod` values are documented on the superclass
+        :meth:`~music21.base.Music21Object.getContextByClass`:
+
+        *    `'getElementBefore'`
+        *    `'getElementAfter'`
+        *    `'getElementAtOrBefore'` (Default)
+        *    `'getElementAtOrAfter'`
+
+        Ask for measure 2's clef context and get the clef at 0.0 in measure 2:
+
+        >>> p = converter.parse('tinyNotation: c1 d1')
+        >>> m = p.measure(2)
+        >>> m.insert(0, clef.BassClef())
+        >>> m.getContextByClass(clef.Clef, getElementMethod='getElementAtOrBefore')
```

Comment:
remove `getElementMethod='getElementAtOrBefore'` to show more clearly that this is the default behavior.